admin/admin-base.php

```diff
@@ -497,4 +499,36 @@ public function admin_bar_menu( $wp_admin_bar ) {
 			);
 		}
 	}
+
+	/**
+	 * Remove the customize submenu when using a block theme.
+	 *
+	 * @since 3.2
+	 *
+	 * @return void
+	 */
+	public function remove_customize_submenu() {
+		// Exit if a block theme isn't activated.
+		if ( ! function_exists( 'wp_is_block_theme' ) || ! wp_is_block_theme() ) {
+			return;
+		}
+
+		global $wp_filter;
+		if ( isset( $wp_filter['customize_register'] ) ) {
+			$customize_register_hooks = count( $wp_filter['customize_register']->callbacks );
+			if ( $customize_register_hooks > 1 ) {
+				return;
+			}
+		}
```

Comment:
We must not remove the sub-menu if there is nothing in `$wp_filter['customize_register']`.

```suggestion
		if ( empty( $wp_filter['customize_register'] ) ) {
			return;
		}

		$customize_register_hooks = count( $wp_filter['customize_register']->callbacks );

		if ( $customize_register_hooks > 1 ) {
			return;
		}
```